lib/ansible/modules/remote_management/hpilo/hpilo_facts.py

```diff
@@ -0,0 +1,245 @@
+    description:
+    - The login name to authenticate to the HP iLO interface.
+    default: Administrator
+  password:
+    description:
+    - The password to authenticate to the HP iLO interface.
+    default: admin
+requirements:
+- hpilo
+notes:
+- This module ought to be run from a system that can access the HP iLO
+  interface directly, either by using C(local_action) or using C(delegate_to).
+'''
+
+EXAMPLES = r'''
+# Task to gather facts from a HP iLO interface only if the system is an HP server
+- hpilo_facts:
+    host: YOUR_ILO_ADDRESS
+    login: YOUR_ILO_LOGIN
+    password: YOUR_ILO_PASSWORD
+  when: cmdb_hwmodel.startswith('HP ')
+  delegate_to: localhost
+
+- fail:
+    msg: 'CMDB serial ({{ cmdb_serialno }}) does not match hardware serial ({{ hw_system_serial }}) !'
+  when: cmdb_serialno != hw_system_serial
+'''
+
+RETURN = r'''
+# Typical output of HP iLO_facts for a physical system
+hw_bios_date:
+    description: BIOS date
+    returned: always
+    type: string
+    sample: 05/05/2011
+
+hw_bios_version:
+    description: BIOS version
+    returned: always
+    type: string
+    sample: P68
+
+hw_ethX:
+    description: Interface information (for each interface)
+    returned: always
+    type: dictionary of information (macaddress)
+    sample:
+      - macaddress: 00:11:22:33:44:55
+        macaddress_dash: 00-11-22-33-44-55
+
+hw_eth_ilo:
+    description: Interface information (for the iLO network interface)
+    returned: always
+    type: dictionary of information (macaddress)
+    sample:
+      - macaddress: 00:11:22:33:44:BA
+      - macaddress_dash: 00-11-22-33-44-BA
+
+hw_product_name:
+    description: Product name
+    returned: always
+    type: string
+    sample: ProLiant DL360 G7
+
+hw_product_uuid:
+    description: Product UUID
+    returned: always
+    type: string
+    sample: ef50bac8-2845-40ff-81d9-675315501dac
+
+hw_system_serial:
+    description: System serial number
+    returned: always
+    type: string
+    sample: ABC12345D6
+
+hw_uuid:
+    description: Hardware UUID
+    returned: always
+    type: string
+    sample: 123456ABC78901D2
+'''
+
+import sys
+import re
+import warnings
+from ansible.module_utils.basic import AnsibleModule
+
+try:
+    import hpilo
+    HAS_HPILO = True
+except ImportError:
+    HAS_HPILO = False
+
+
+# Suppress warnings from hpilo
+warnings.simplefilter('ignore')
+
+
+def parse_flat_interface(entry, non_numeric='hw_eth_ilo'):
+    try:
+        factname = 'hw_eth' + str(int(entry['Port']) - 1)
+    except:
+        factname = non_numeric
+
+    facts = {
+        'macaddress': entry['MAC'].replace('-', ':'),
+        'macaddress_dash': entry['MAC']
+    }
+    return (factname, facts)
+
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec = dict(
+            host = dict(required=True, type='str'),
+            login = dict(default='Administrator', type='str'),
+            password = dict(default='admin', type='str', no_log=True),
+        )
```

Comment:
Please add `supports_check_mode=True`. With this seeting, facts modules will also be executed  in `--check` mode.

~~~ python
    module = AnsibleModule(
        argument_spec = dict(
            host=dict(required=True, type='str'),
            login=dict(default='Administrator', type='str'),
            password=dict(default='admin', type='str', no_log=True),
        ),
     supports_check_mode=True,
    )
~~~